src/unix/fs.c

```diff
@@ -120,7 +120,9 @@
     }                                                                         \
     else {                                                                    \
       uv__fs_work(&req->work_req);                                            \
-      return req->result;                                                     \
+      /* Saturate return value at INT32_MAX, to avoid confusion with error return codes.    */ \
+      /* The real result value should be read from req->result whenever it might be larger. */ \
+      return (req->result < INT32_MAX ? req->result : INT32_MAX);             \
```

Comment:
Well, it will only affects the ones where we are returning the wrong answer now. This alters it so that the answer may still be wrong (it's not the only place where we assume – per the C standard – that size_t fits into a ssize_t and is smaller than half of memory), but it seemed like this at least would permit existing error handling code (testing for return < 0) to work correctly on 64-bit systems now.

Reply:
For another PR, it might me a good idea state more clearly in the docs that the correct way to get the result of sync requests is using `uv_fs_get_result()` and not checking the return value of the function (except for errors).